Dockerfile

```diff
@@ -92,134 +109,100 @@ ARG VERSION_PKG="github.com/talos-systems/talos/internal/pkg/version"
+# The osctl targets build the osctl binaries.
 
-FROM base AS proxyd-build
+FROM base AS osctl-linux-build
 ARG SHA
 ARG TAG
 ARG VERSION_PKG="github.com/talos-systems/talos/internal/pkg/version"
-WORKDIR /src/internal/app/proxyd
-RUN --mount=type=cache,target=/root/.cache go build -a -ldflags "-s -w -X ${VERSION_PKG}.Name=Server -X ${VERSION_PKG}.SHA=${SHA} -X ${VERSION_PKG}.Tag=${TAG}" -o /proxyd
-RUN chmod +x /proxyd
-
-FROM scratch AS proxyd
-COPY --from=proxyd-build /proxyd /proxyd
-ENTRYPOINT ["/proxyd"]
-
-# The ntpd target builds the ntpd image.
+WORKDIR /src/cmd/osctl
+RUN GOOS=linux GOARCH=amd64 go build -a -ldflags "-s -w -X ${VERSION_PKG}.Name=Client -X ${VERSION_PKG}.SHA=${SHA} -X ${VERSION_PKG}.Tag=${TAG}" -o /osctl-linux-amd64
+RUN chmod +x /osctl-linux-amd64
+FROM scratch AS osctl-linux
+COPY --from=osctl-linux-build /osctl-linux-amd64 /osctl-linux-amd64
 
-FROM base AS ntpd-build
+FROM base AS osctl-darwin-build
 ARG SHA
 ARG TAG
 ARG VERSION_PKG="github.com/talos-systems/talos/internal/pkg/version"
-WORKDIR /src/internal/app/ntpd
-RUN --mount=type=cache,target=/root/.cache go build -a -ldflags "-s -w -X ${VERSION_PKG}.Name=Server -X ${VERSION_PKG}.SHA=${SHA} -X ${VERSION_PKG}.Tag=${TAG}" -o /ntpd
-RUN chmod +x /ntpd
-
-FROM scratch AS ntpd
-COPY --from=ntpd-build /ntpd /ntpd
-ENTRYPOINT ["/ntpd"]
-
-# The binaries target allows for parallel compilation of all binaries.
-
-FROM scratch AS binaries-build
-COPY --from=init / /
-COPY --from=osd / /
-COPY --from=trustd / /
-COPY --from=proxyd / /
-COPY --from=ntpd / /
-COPY --from=osctl-linux-amd64 / /
-COPY --from=osctl-darwin-amd64 / /
-
-FROM scratch AS binaries
-COPY --from=binaries-build /osctl-linux-amd64 /osctl-linux-amd64
-COPY --from=binaries-build /osctl-darwin-amd64 /osctl-darwin-amd64
+WORKDIR /src/cmd/osctl
+RUN GOOS=darwin GOARCH=amd64 go build -a -ldflags "-s -w -X ${VERSION_PKG}.Name=Client -X ${VERSION_PKG}.SHA=${SHA} -X ${VERSION_PKG}.Tag=${TAG}" -o /osctl-darwin-amd64
+RUN chmod +x /osctl-darwin-amd64
+FROM scratch AS osctl-darwin
+COPY --from=osctl-darwin-build /osctl-darwin-amd64 /osctl-darwin-amd64
 
 # The kernel target is the linux kernel.
 
-ARG KERNEL_IMAGE
-FROM ${KERNEL_IMAGE} as kernel
-
-# The initramfs target creates the compressed initramfs.
-
-FROM base AS init-build
-ARG SHA
-ARG TAG
-ARG VERSION_PKG="github.com/talos-systems/talos/internal/pkg/version"
-WORKDIR /src/internal/app/init
-RUN --mount=type=cache,target=/root/.cache go build -a -ldflags "-s -w -X ${VERSION_PKG}.Name=Talos -X ${VERSION_PKG}.SHA=${SHA} -X ${VERSION_PKG}.Tag=${TAG}" -o /init
-RUN chmod +x /init
-
-FROM scratch AS init
-COPY --from=init-build /init /init
-
-ARG INITRAMFS_IMAGE
-FROM ${INITRAMFS_IMAGE} AS initramfs-build
-WORKDIR /
-COPY --from=init-build /init /init
-
-ARG TOOLCHAIN_IMAGE
-FROM ${TOOLCHAIN_IMAGE} AS initramfs-archive
-COPY --from=initramfs-build / /initramfs
+FROM scratch AS kernel
+COPY --from=docker.io/autonomy/kernel:2ac99a0 /boot/vmlinuz /vmlinuz
```

Comment:
nit, should modules be included with this?

Reply:
Reason we don't do that is because we use the `local` exporter to drop the `vmlinuz` artifact into `./build`, and add the modules here will mean that they get exported locally too.